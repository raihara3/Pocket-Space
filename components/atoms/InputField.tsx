import React, { memo } from 'react'
import styled from 'styled-components'

interface Props {
  id?: string
  name?: string
  ref?: any
  type?: 'text' | 'number' | 'password'
  readonly?: boolean
  disabled?: boolean
  placeholder?: string
  defaultValue?: string
  onChange: (e: any) => void
}

const InputField: React.FC<Props> = ({
  id,
  name,
  ref,
  type = 'text',
  readonly = false,
  disabled = false,
  placeholder,
  defaultValue,
  onChange
}) => {
  return (
    <Input
      id={id}
      name={name}
      ref={ref}
      type={type}
      readOnly={readonly}
      disabled={disabled}
      placeholder={placeholder}
      defaultValue={defaultValue}
      onChange={(e) => onChange(e)}
    />
  )
}

const Input = styled.input<{readOnly: boolean}>`
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  resize: none;
  width: 100%;
  height: 1.5rem;
  margin: 0 0 10px;
  padding: 0 0 8px;
  border: 0;
  outline: none;
  background: transparent;
  color: #ffffff;
  border-bottom: 1px solid rgba(255,255,255,0.3);

  ${({readOnly}) => readOnly && `
    color: rgba(255,255,255,0.3);
    font-size: 10px;
  `}
`

export default memo(InputField)
