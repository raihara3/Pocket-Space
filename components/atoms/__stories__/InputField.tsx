import React from 'react'
import InputField from '../InputField'

export const normal = () => {
  return (
    <InputField
      placeholder='Input field'
    />
  )
}

export const readOnly = () => {
  return (
    <InputField
      placeholder='Input field'
      readonly={true}
    />
  )
}

export const disabled = () => {
  return (
    <InputField
      placeholder='disabled'
      disabled={true}
    />
  )
}

export const hasError = () => {
  return (
    <InputField
      placeholder='Input field'
      hasError={true}
      errorMessage='Error!!'
    />
  )
}

export default {
  title: 'InputField'
}
