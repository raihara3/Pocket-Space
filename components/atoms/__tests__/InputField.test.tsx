import React from 'react'
import renderer from 'react-test-renderer'
import InputField from '../InputField'

test('normal', () => {
  const tree = renderer
    .create(<InputField placeholder="Input field" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
test('readOnly', () => {
  const tree = renderer
    .create(<InputField placeholder="Input field" readonly={true} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
test('disabled', () => {
  const tree = renderer
    .create(<InputField placeholder="Input field" disabled={true} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
test('hasError', () => {
  const tree = renderer
    .create(
      <InputField
        placeholder="Input field"
        hasError={true}
        errorMessage="has error!"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
