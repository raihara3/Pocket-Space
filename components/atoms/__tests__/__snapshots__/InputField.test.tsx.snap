// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disabled 1`] = `
<div
  className="InputField__InputBox-xsvhdm-0 jMqyAe"
>
  <label>
    <input
      className="InputField__Input-xsvhdm-1 dmqiXI"
      disabled={true}
      onChange={[Function]}
      placeholder="Input field"
      type="text"
    />
  </label>
</div>
`;

exports[`hasError 1`] = `
<div
  className="InputField__InputBox-xsvhdm-0 jMqyAe"
>
  <label>
    <input
      className="InputField__Input-xsvhdm-1 dmqiXI"
      disabled={false}
      onChange={[Function]}
      placeholder="Input field"
      type="text"
    />
    <span
      className="InputField__ErrorMessage-xsvhdm-2 jRXaMi"
    >
      has error!
    </span>
  </label>
</div>
`;

exports[`normal 1`] = `
<div
  className="InputField__InputBox-xsvhdm-0 jMqyAe"
>
  <label>
    <input
      className="InputField__Input-xsvhdm-1 dmqiXI"
      disabled={false}
      onChange={[Function]}
      placeholder="Input field"
      type="text"
    />
  </label>
</div>
`;

exports[`readOnly 1`] = `
<div
  className="InputField__InputBox-xsvhdm-0 jMqyAe"
>
  <label>
    <input
      className="InputField__Input-xsvhdm-1 jmKEkY"
      disabled={false}
      onChange={[Function]}
      placeholder="Input field"
      type="text"
    />
  </label>
</div>
`;
